api/auth.go

```diff
@@ -70,6 +72,7 @@ func WithToken(token string) Option {
 	return clientFunc(func(c *Client) error {
 		c.log.Debug("setting up auth", zap.String("token", token))
 		c.auth.token = token
+		c.auth.expiresAt = time.Now().Add(DefaultTokenExpiryTime * time.Second)
```

Comment:
Now that I think about this, there is no way we can know when will the token expire. 🤔